api/routes/otp/controllers.js

```diff
@@ -0,0 +1,37 @@
+const DB = require('models');
+const ApiError = require('base/error');
+const BaseDetailController = require('base/controllers/detailController');
+const { randomGenerator } = require('utils/random');
+
+class OtpLoginController extends BaseDetailController {
+  model = DB.users;
+
+  getObject() {
+    return this.model.findOne({
+      where: {
+        phone: this.request.body.phone,
+      },
+      include: this.generateIncludeClause(),
+    });
+  }
+
+  async post() {
+    const user = await this.getObject();
+    if (!user) {
+      throw ApiError('User not found');
+    }
+    const redisClient = this.app.getService('redis').client;
```

Comment:
This defeats the purpose of abstraction. If we access the client directly we loose the ease of having an in-repo interface for the methods. Better would be to write a basic method like
`function set(key, value, opts = { ttl: 60*1 })`